be/src/exec/pipeline/pipeline_builder.cpp

```diff
@@ -64,11 +64,14 @@ OpFactories PipelineBuilderContext::maybe_interpolate_local_shuffle_exchange(
 }
 
 OpFactories PipelineBuilderContext::gather_pipelines_to_one(std::vector<OpFactories>& pred_operators_list) {
```

Comment:
What' the difference `gather_pipelines_to_one` with `maybe_interpolate_local_passthrough_exchange `

Reply:
`maybe_interpolate_local_passthrough_exchange()` merge outputs of multiple **drivers** of one pred pipeline into a driver of the post pipeline. It is used when the post operator cannot be parallelized and the pred operator can be parallelized. If the post operator and pred operator in a same pipeline, then the DOP of this pipeline can only be one. By using this method, we can divide the pred and post operator into two different pipelines.

`gather_pipelines_to_one()` merge outputs of multiple pred **pipeline** into a post pipeline. For example, UnionNode receives data from multiple child nodes, which belong to different pipelines.